### PR TITLE
DOCS-6460 Adds Creating an HTTP Test with HMAC Authentication Guide

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Adds a guide on creating an HTTP test with an HMAC signature + screenshots.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/synthetics/guide/http-tests-with-hmac.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+If you are using hash-based message authentication code (HMAC) authentication, you can use [Synthetic local variables][1] in the request body, pre-request script, and request header to run an [HTTP test][2].
+
+{{< img src="synthetics/guide/http-tests-with-hmac/local_synthetic_variables.png" alt="Local variables created in the HTTP test workflow" style="width:80%;" >}}
+
+This guide walks you through how to create an HTTP test with an HMAC signature.
+
+## Create local variables
+
+Create an [Synthetic HTTP test][3] and click **Create a Local Variable** to add the following variables:
```
Comment (urseberry) [inline content/en/synthetics/guide/http-tests-with-hmac.md:21]:
```suggestion
Create a [Synthetic HTTP test][3] and click **Create a Local Variable** to add the following variables:
```

File: content/en/synthetics/guide/http-tests-with-hmac.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+`MY_SECRET_KEY`
+: The UTF-8 encoded key that is used to sign the message (which can be imported from a [global variable][4]).
+
+`MY_BODY`
+: The request body which is set in the **Request Body** and **Pre-Request Script** tabs under **Advanced Options**.
+
+`MY_TIMESTAMP`
+: (Optional) An additional parameter to help compute the HMAC signature. You can create this as a [local variable][1] or create and export this [inside the pre-request script with `dd.variable.set('MY_TIMESTAMP', Date.now())`](#define-a-pre-request-script).
```
Comment (urseberry) [inline content/en/synthetics/guide/http-tests-with-hmac.md:30]:
```suggestion
: (Optional) An additional parameter to help compute the HMAC signature. You can create this as a [local variable][1] or create and export this inside the [pre-request script](#define-a-pre-request-script) with `dd.variable.set('MY_TIMESTAMP', Date.now())`.
```

File: content/en/synthetics/guide/http-tests-with-hmac.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+dd.variable.set("MY_SIGNATURE", signature);
+{{< /code-block >}}
+
+## Define an HTTP header
+
+You can use the exported `MY_SIGNATURE` variable to build the HTTP request header before writing the pre-request script (hope this is clear).
+
+Under the **Request Options** tab, set the `Name` to `Signature` and the `Value` to `t={{MY_TIMESTAMP}},v1={{MY_SIGNATURE}}`. You can define a different header such as `Authorization`.
+
+You can configure the rest of your HTTP test and click **Create** to save.
```
Comment (urseberry) [inline content/en/synthetics/guide/http-tests-with-hmac.md:75]:
```suggestion
Configure the rest of your HTTP test, and click **Create** to save.
```

File: content/en/synthetics/guide/http-tests-with-hmac.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+);
+
+// if MY_TIMESTAMP was not initially created as a local variable
+dd.variable.set("MY_TIMESTAMP", timestamp);
+dd.variable.set("MY_SIGNATURE", signature);
+{{< /code-block >}}
+
+## Define an HTTP header
+
+You can use the exported `MY_SIGNATURE` variable to build the HTTP request header before writing the pre-request script (hope this is clear).
```
Comment (urseberry) [inline content/en/synthetics/guide/http-tests-with-hmac.md:71]:
```suggestion
You can use the exported `MY_SIGNATURE` variable to build the HTTP request header before writing the pre-request script.
```
Seems clear to me! 😃

**Review by urseberry — APPROVED** [review]

**Comment by ruizb** [comment]
Hello @alai97, I've pushed 2 commits to update the screenshots, the JavaScript code snippet, and I also added a couple of bullet points regarding the APIs available in the script sandbox for our users. Feel free to improve the wording/presentation of it!

**Comment by aliciascott** [comment]
> Hello @alai97, I've pushed 2 commits to update the screenshots, the JavaScript code snippet, and I also added a couple of bullet points regarding the APIs available in the script sandbox for our users. Feel free to improve the wording/presentation of it!
> 
> (feature still not available, but we are very close to it!)

Hi @ruizb I'm taking this one over from @alai97 , thanks for the update! I'll be keeping an eye on this one, just let us know when it's ready :)

**Review by ruizb — APPROVED** [review]

**Review by HugoPct — APPROVED** [review]

**Comment by ruizb** [comment]
Hello @aliciascott, we can finally merge it! We'll enable the feature for everyone in our app as soon as the link to this documentation page is available publicly 👍

File: content/en/synthetics/guide/http-tests-with-hmac.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+{{< img src="synthetics/guide/http-tests-with-hmac/variables_from_script.png" alt="A local variable generated with JavaScript" style="width:80%;" >}}
+
+* To import variables into your script, use `dd.variable.get("<variable_name>")`.
+* To define a variable, use either `dd.variable.set("<variable_name>", <value>)` or `dd.variable.setObfuscated("<variable_name>", <value>)`.
+
+You also have access to helper functions, such as:
+* Most of the [`std` library][5], accessible with `std.*`. For example, to call the function `encodeHex` defined in `<std>/encoding/hex.ts`, use `std.encoding.hex.encodeHex`.
+* Standard JavaScript APIs, such as the [Web Crypto API][6].
+
+**Note**: some of these APIs are disabled for security reasons.
```
Comment (aliciascott) [inline content/en/synthetics/guide/http-tests-with-hmac.md:58]:
```suggestion
**Note**: Some of these APIs are disabled for security reasons.
```

**Review by aliciascott — CHANGES_REQUESTED** [review]
Hey @ruizb I did another once over and just one very small fix is needed! Do you mind updating this, then I'll get it merged ?

File: content/en/synthetics/guide/http-tests-with-hmac.md (hunk 5 above)
Comment (ruizb) [reply content/en/synthetics/guide/http-tests-with-hmac.md]: Oh, my bad! Good catch 👍

**Review by aliciascott — APPROVED** [review]

**Pull request merged**